app/src/main/java/org/fossasia/openevent/general/event/EventsRecyclerAdapter.kt

```diff
@@ -34,6 +36,44 @@ class EventsRecyclerAdapter : RecyclerView.Adapter<EventsRecyclerAdapter.EventVi
                         .placeholder(R.drawable.ic_launcher_background)
                         .into(itemView.eventImage)
             }
+
+            itemView.shareFab.setOnClickListener{
+                val eventName = event.name
+                val description = if (event.description != "") event.description else "Unavailable"
+                val startDate = dateFormat(event.startsAt)
+                val endDate = dateFormat(event.endsAt)
+                val startTime = timeFormat(event.startsAt)
+                val endTime = timeFormat(event.endsAt)
+                val eventUrl = if (event.externalEventUrl != null) Uri.parse(event.externalEventUrl) else "Unavailable"
+
+                val sendIntent = Intent()
+                sendIntent.action = Intent.ACTION_SEND
+                val message = String.format(itemView.resources.getString(R.string.event_name) +
+                              "\n\n" + itemView.resources.getString(R.string.event_description) +
+                              "\n\n" + itemView.resources.getString(R.string.starts_on) +
+                              "\n" + itemView.resources.getString(R.string.start_time)  +
+                              "\n\n" + itemView.resources.getString(R.string.ends_on) +
+                              "\n" + itemView.resources.getString(R.string.end_time)  +
+                              "\n\n" + itemView.resources.getString(R.string.event_link) ,
+                              eventName, description, startDate, startTime, endDate, endTime, eventUrl)
+                sendIntent.putExtra(Intent.EXTRA_TEXT, message)
```

Comment:
what if timeFormat function generates empty string, in this case you are still appending it in the share text.
Only send the information available, check if the field is empty and dont send in that case

Reply:
Yup :+1:

Reply:
best would be to wait for the mentor for his views

Reply:
Yes, only send present information. Also, make use of StringBuilder rather than concatenation

Reply:
Sure